cmd/gokeyless/gokeyless.go

```diff
@@ -122,6 +141,38 @@ func LoadKey(in []byte) (priv crypto.Signer, err error) {
 	return derhelpers.ParsePrivateKeyDER(in)
 }
 
+func validCert(cert *x509.Certificate) bool {
+	now := time.Now()
+	if now.Before(cert.NotBefore) {
+		return false
+	}
+
+	if now.After(cert.NotAfter) {
+		return false
+	}
+
+	// certificate expires in a month
+	if now.Add(time.Hour * 24 * 30).After(cert.NotAfter) {
+		return false
+	}
+
+	return true
+}
+func needNewCertAndKey() bool {
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		log.Errorf("cannot load server cert/key: %v", err)
+		return true
+	}
+
+	if !validCert(cert.Leaf) {
+		log.Errorf("certificate is either not yet valid, expired or expiring in a month")
```

Comment:
Can we be more precise and show the actual cause - not valid/expired is different from expiring in a month (and I'm not sure we should force failure if hasn't *actually* expired yet)